RealmSwift/Sync.swift

```diff
@@ -530,3 +553,490 @@ extension Realm {
         return SyncSession(for: rlmRealm)
     }
 }
+
+#if canImport(Combine)
+@available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, macCatalyst 13.0, macCatalystApplicationExtension 13.0, *)
+public extension App {
+    /// :nodoc:
+    enum UserError: Error {
+        /// UserError.uncertainState - Fallback for optional errors  in Future calls. Should never happened while it's still a technically possible case.
+        case uncertainState
+    }
+
+    /**
```

Comment:
Will "/**" type of comments be picked up by the doc-tool?
Is there a way to inherit this documentation from non combine version of the methods?

Reply:
Docstrings are supported (though triple slash `///` is preferred).

Swift has no `@inheritdoc` (or anything similar) like Java does, unfortunately.

Reply:
Too bad there in no inherit. (we should make that..)
How do we ensure these docs are kept in sync? Would it be better to locate similar methods next to each other instead?

Reply:
At least the comment style is inconsistent :-)

Reply:
In this case, locating the similar methods next to each other would create a lot of noise around the `@available` attribute and `#if canImport(...)` macros. Having an extension containing all this functionality makes it significantly better for user discoverability as well.

The best solution here (both to make sure docs are kept in sync and for the sake of organisation) is to break up `Sync.swift` and `App.swift` and put all the extensions in their own files (`User.swift`, `MongoCollection.swift`, etc.).

Originally, there wasn't enough being extending to warrant the creation of all these files, but we've definitely reached that point.

I'll add a ticket to quick wins.

Reply:
We can add a reference in the documentation to the alternative methods

Reply:
Could do, but I'm not sure SwiftLint will approve of that. We'll have to see.